packages/blade-mcp/src/tools/getChangelog.ts

```diff
@@ -0,0 +1,112 @@
+import type { ToolCallback } from '@modelcontextprotocol/sdk/server/mcp.js';
+import dedent from 'dedent';
+import { z } from 'zod';
+import { handleError, sendAnalytics } from '../utils/analyticsUtils.js';
+import {
+  getLatestVersion,
+  getRangeChangelogs,
+  stringifyChangelog,
+} from '../utils/changelogParser.js';
+import { analyticsToolCallEventName } from '../utils/tokens.js';
+
+const getChangelogToolName = 'get_blade_changelog';
+
+const getChangelogToolDescription = `
+Get the changelog of blade to help consumer with blade upgrades and checking release notes. Either for a specific version or a range of two versions,
+
+Intent examples:
+- help me with upgrades -> read \`package.json\` to get consumer's blade version to use it as \`fromVersion\` and then use \`toVersion\` as \`latest\` version (range)
+- what was changed in x.x.x version? -> Just \`fromVersion\` (set isRange to false & omit toVersion)
+- what was changed between x.x.x and y.y.y versions? -> \`fromVersion\` and \`toVersion\` (set isRange to true)
+- what is the latest version of blade? -> \`fromVersion\` as \`latest\` (set isRange to false & omit toVersion)
+`;
+
+const getChangelogToolSchema = {
+  fromVersion: z
+    .string()
+    .describe('The start version of the range to get the changelog for. E.g. 10.15.0'),
+  toVersion: z
+    .string()
+    .optional()
+    .describe('The end version of the range to get the changelog for. E.g. 10.16.0'),
+  isRange: z
+    .boolean()
+    .optional()
+    .describe(
+      'Whether to get the changelog for a range of versions or a specific version, this is based on intent of user.',
+    ),
+};
+
+const getChangelogToolCallback: ToolCallback<typeof getChangelogToolSchema> = async ({
+  fromVersion,
+  toVersion,
+  isRange,
+}) => {
+  try {
+    const changelogURL =
+      'https://raw.githubusercontent.com/razorpay/blade/refs/heads/master/packages/blade/CHANGELOG.md';
+    const response = await fetch(changelogURL, {
```

Comment:
need to check if this works with zscaler in consumer apps otherwise we can use our server as proxy

Reply:
i've tested this, works. github won't load if githubusercontent is blocked by zscaler :p